packages/playground/website/src/lib/query-api.ts

```diff
@@ -0,0 +1,110 @@
+import { Blueprint } from '@wp-playground/blueprints';
+import { makeBlueprint } from './make-blueprint';
+import { type SiteInfo } from './site-storage';
+
+const defaultPhpVersion = '8.0';
+
+/**
+ * Get a URL with query params representing the specified site settings.
+ *
+ * @param baseUrl The current Playground URL
+ * @param siteInfo The site info to convert to Playground query params
+ * @returns URL with query params representing the site info
+ */
+export function siteInfoToUrl(baseUrl: URL, siteInfo: SiteInfo): URL {
```

Comment:
Aha! We've hit the runtime configuration vs site configuration problem (see ["Runtime setup options"](https://github.com/WordPress/blueprints-library/issues/6)). Let's work through this methodically and avoid just translating all the Blueprint options to query args. Instead, let's make this v1 of the runtime configuration interface. 

Let's treat things like languages, themes, plugins, multisite etc. as build-time features that are only relevant when the site is first created. If we tried re-applying them to an existing site, we'd end up with, say, two copies of the same theme.

That leaves us with settings that don't trigger any filesystem– or database–level updates in `/wordpress`. For sure that's networking, PHP version, and landing page. Probably also `login`, even though it causes some database writes. What else?

Reply:
I wonder what would be the ripple effect of decoupling the runtime options across JS API, Blueprints API, Query API, if any.

For example, imagine a version of the Query API that distinguishes between "site build options" and "runtime options":

https://playground.wordpress.net/?build[plugin]=gutenberg&runtime[networking]=yes

These long names would be difficult to type and to remember so our current short parameters (`?plugin=gutenberg&networking=yes`) seem to be a more useful choice.

However, Blueprints and JS API are quite verbose already so maybe there may be an opportunity for an API refresh in there. If not, perhaps the documentation could distinguish between the two types of configuration options 🤔

Reply:
Also, notice this translates Blueprint->Query args only for Playground to do Query args->Blueprint. What if we passed just a Blueprint? Or even just a site slug, since we know the config for each slug?

Reply:
It would be highly useful to have stable site URLs. Say I go to playground.wordpress.net/my–base–theme and it always brings up the same site. Also, the site itself wouldn't use a random numeric scope, but the slug

Reply:
Then we could pair that with github autocommit and offer public site urls, e.g. public.playground.wordpress.net/adamziel/blog

Reply:
And that could later form v1 of collaboration and site sync. Cc @akirk @griffbrad @dawidurbanski @dmsnell